php/libraries/DataFrameworkMenu.class.inc

```diff
@@ -101,19 +110,23 @@ abstract class DataFrameworkMenu extends NDB_Menu_Filter
     {
         $provisioner = $this->getBaseDataProvisioner();
 
-        $user = \NDB_Factory::singleton()->user();
-
         // UserSiteMatch throws an exception if used on a model without a getCenterID
         // or getCenterIDs function, so we only apply it on sub-classes that
         // specifically say they support it.
-        if ($this->useSiteFilter()
-            && $user->hasPermission("access_all_profiles") == false
-        ) {
+        if ($this->useSiteFilter()) {
```

Comment:
I don't see what the difference between `useSiteFilter` and `useSitePermissionsFilter` is supposed to be. Why doesn't this just use something like a `allSitePermissionNames` function and update the if statement to check that, always using the `HasAnyPermissionOrUserSiteMatch` filter?